.github/workflows/python.yml

```diff
@@ -30,25 +33,116 @@ jobs:
         environment-file: .github/python_test_env.yml
 
     - name: Install
-      run: pip install . -v
+      run: python -m pip install . -v
 
     - name: Test
-      run: pytest --cov exactextract --cov-report lcov python/tests
+      run: python -m pytest --cov exactextract --cov-report lcov python/tests
 
     - name: Upload coverage report
       uses: codecov/codecov-action@v3
       with:
         files: coverage.lcov
         token: ${{ secrets.CODECOV_TOKEN }}
-      
+
     - name: Build source distribution
       if: matrix.os == 'ubuntu-latest' && matrix.python-version == '3.11'
       run: python -m build . --sdist
 
     - name: Upload distribution(s)
-      if: matrix.os == 'ubuntu-latest' && matrix.python-version == '3.11'
+      if: startsWith(github.event.ref, 'refs/tags') || github.event_name == 'release'
```

Comment:
What would be the usual way to upload pre-release wheels/sdist for testing? Would it be to add an option to manually invoke the workflow (as I did in the wheel build) or is it better to create "pre-release" release?

Reply:
They are uploaded as artifact to the github action (or should be, but that seems to be failing at the moment), and then you _can_ download them from there and install locally from the downloaded wheel.

That's not a super easy workflow, but at least makes it possible to have someone test those wheels.

You could upload them to PyPI as a pre-release wheels at some point, but for that it might be better to do a beta release or something.

Reply:
Ah, the artifacts are not showing up on the latest commit because the wheels builds were disabled now. Looking at one of the older commits, you can get the artifacts from https://github.com/isciences/exactextract/actions/runs/8234420621